inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java

```diff
@@ -0,0 +1,240 @@
+        userId.setDescription("Enter your Inwebo UserName");
+        configProperties.add(userId);
+
+        Property serviceId = new Property();
+        serviceId.setName(InweboConstants.SERVICE_ID);
+        serviceId.setDisplayName("Service Id");
+        serviceId.setRequired(true);
+        serviceId.setDescription("Enter your service Id");
+        configProperties.add(serviceId);
+
+        Property p12file = new Property();
+        p12file.setName(InweboConstants.INWEBO_P12FILE);
+        p12file.setDisplayName("Certificate File");
+        p12file.setRequired(true);
+        p12file.setDescription("Enter your p12_file path");
+        configProperties.add(p12file);
+
+        Property p12password = new Property();
+        p12password.setName(InweboConstants.INWEBO_P12PASSWORD);
+        p12password.setDisplayName("Certificate Password");
+        p12password.setRequired(true);
+        p12password.setConfidential(true);
+        p12password.setDescription("Enter your p12_password");
+        configProperties.add(p12password);
+
+        Property retryCount = new Property();
+        retryCount.setName(InweboConstants.RETRYCOUNT);
+        retryCount.setDisplayName("Retry Count");
+        retryCount.setRequired(true);
+        retryCount.setDescription("Number of attempts waiting for authentication(<10)");
+        configProperties.add(retryCount);
+
+        Property retryInterval = new Property();
+        retryInterval.setName(InweboConstants.RETRINTERVAL);
+        retryInterval.setDisplayName("Retry Interval");
+        retryInterval.setRequired(true);
+        retryInterval.setDescription("Period of time for waiting eg:1000");
+        configProperties.add(retryInterval);
+        return configProperties;
+
+    }
+
+    /**
+     * Process the response of the Inwebo end-point
+     */
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+        if (pushResponse.contains(InweboConstants.PUSHRESPONSE)) {
+            log.info("Authentication Successful");
+            context.setSubject("Success");
+            Map<String, Object> userClaims = getUserClaims();
+            if (userClaims != null && !userClaims.isEmpty()) {
+                context.setSubjectAttributes(getSubjectAttributes(userClaims));
+                context.setSubject("Authentication Success");
+            } else {
+                throw new AuthenticationFailedException("Selected user profile found");
+            }
+        } else
+            throw new AuthenticationFailedException("Authentication failed");
+        pushResponse = null;
+    }
+
+    protected Map<ClaimMapping, String> getSubjectAttributes(
+            Map<String, Object> claimMap) {
+
+        Map<ClaimMapping, String> claims = new HashMap<ClaimMapping, String>();
```

Comment:
From Java - 1.7 onwards we can get rid of specifiying the generics at initiation.. 
new HashMap<ClaimMapping, String>()  --> new HashMap<>()
https://docs.oracle.com/javase/8/docs/technotes/guides/language/type-inference-generic-instance-creation.html